src/adapters/net.cpp

```diff
@@ -248,6 +253,10 @@ namespace net {
       return false;
     }
 
+    if(m_bridge) { // If bridge network then link speed cannot be computed TODO: Identify the physical network in bridge and compute the link speed
```

Comment:
This is not what I meant. We still need to query the interface with `ioctl`, but we only set `m_linkspeed` if we are not using a bridge.

Reply:
We are already checking the connected status as a part of the call to update 
`
m_connected = network->connected();
`

And above call makes use of `ioctl` call to check the status and it works in case of bridge interface.

If we compute the link speed on bridge interface the call fails and false is returned

`
if (ioctl(*m_socketfd, SIOCETHTOOL, &request) == -1) {
      return false;
    }
`

Did you mean to return true from above else block in case of bridge interface?

Reply:
Ah, I see, I though we needed that `ioctl` to also check if the interface works properly not just to get the linkspeed.

Say, what is the content of `/sys/class/net/br0/speed` on your system?

Reply:
Please add this to the "if-else-if" block at the beginning. 

```suggestion
    } else if (m_bridge) { 
    /* If bridge network then link speed cannot be computed 
     * TODO: Identify the physical network in bridge and compute the link speed
     */
```

Reply:
Hi @patrick96 , I am not sure if you mean this

```suggestion
if (m_tuntap) {
      return true;
    } else if(m_bridge) { // If bridge network then link speed cannot be computed TODO: Identify the physical network in bridge and compute the link speed
      return true;
    } else if (!network::query(accumulate)) {
      return false;
    } 
```

If we insert if-else-if as above `network::query(accumulate)` would never get executed. This call calculates the real time upload and download speed for the interface and it works fine for bridge interface.

We need `network::query(accumulate)` to be executed because this gives some additional information about the current network activity. That's why I have kept below if statement separate after we execute this call.

```
if(m_bridge) { // If bridge network then link speed cannot be computed TODO: Identify the physical network in bridge and compute the link speed
      return true;
}
```

Reply:
Please find contents of cat /sys/class/net/br0/speed

$ cat /sys/class/net/br0/speed
cat: /sys/class/net/br0/speed: Invalid argument

Reply:
>  cat: /sys/class/net/br0/speed: Invalid argument

Seems this is consistent with ethtool

> If we insert if-else-if as above network::query(accumulate)

No, I meant
```
if m_tuntap
elseif !network::query
elseif m_bridge
```

Reply:
If I add above if-else-if block, the second if block ‘!network::query’ succeeds and it stops returning true in case of bridge interface.  Which proceeds with link speed computation using ‘ioctl’ and fails returning false.

Thats why I have added it as a separate if statement to return true and stop link speed computation

Reply:
:man_facepalming: you're right. I knew there was some flaw in my logic ^^

But please put the comment on a separate line